src/service/user-service.js

```diff
@@ -0,0 +1,161 @@
+   * @param option {{ page: number, limit: number }}
+   * @param logger { Pino }
+   * @return {{ docs: Array, total: number }}
+   */
+  getUser: async function(query, option, logger = log) {
+    const _log = logger.child({ module: 'user-service', method: 'getUser' });
+    _log.info('retrieve user, query: %j, option: %j', query, option);
+
+    const dbConnection = mongoSvc.getMongoConnection(_log);
+    if (dbConnection == null) {
+      _log.error('Cannot get mongodb connection.');
+      throw new errors.InternalServerError('Cannot get mongodb connection.');
+    }
+
+    try {
+      let q = QueryBuilder().id(query.id).name(query.name).build();
+      const userModel = User.getUserModel(dbConnection);
+      const { docs, total } = await userModel.paginate(q, option);
+      _log.info(`got total ${total} records.`);
+      _log.debug('docs: %j', docs);
+
+      return { docs, total };
+    } catch (err) {
+      _log.error('retrieve user error: %j', errors.fullStack(err));
+      throw new errors.InternalServerError(`Failed to retrieve user: ${err.message}`);
+    }
+  },
+
+  /**
+   * insert user doc to db
+   * @param body {{ id: String, name: String, dob: String, address: String, description: String }}
+   * @param logger { Pino }
+   * @return {{ docs: Array }}
+   */
+  createUser: async function(body, logger = log) {
+    const _log = logger.child({ module: 'user-service', method: 'createUser' });
+    _log.info('create user: %j', body);
+
+    const user = _.pick(body, ['id', 'name', 'dob', 'address', 'description']);
+    _.checkRequired(user, ['id', 'name']);
+
+    const dbConnection = mongoSvc.getMongoConnection(_log);
+    if (dbConnection == null) {
+      _log.error('Cannot get mongodb connection.');
+      throw new errors.InternalServerError('Cannot get mongodb connection.');
+    }
+
+    try {
+      const userModel = User.getUserModel(dbConnection);
+      const docs = await userModel.create([user]);
+      _log.info('user doc saved.');
+      _log.debug('docs: %j', docs);
+
+      return { docs };
+    } catch (err) {
+      _log.error('create user error: %j', errors.fullStack(err));
+      throw new errors.InternalServerError(`Failed to create user: ${err.message}`);
+    }
+  },
+
+  /**
+   * update user doc
+   * @param id { String }
+   * @param body {{ name: String, dob: String, address: String, description: String }}
+   * @param logger { Pino }
+   * @return {{ matchedCount: number, modifiedCount: number, acknowledged: boolean, upsertedId: string, upsertedCount: number }}
+   */
+  updateUser: async function(id, body, logger = log) {
+    const _log = logger.child({ module: 'user-service', method: 'updateUser' });
+    _log.info('update user, id: %s, body: %j', id, body);
+
+    if (_.isEmpty(id)) {
+      _log.error('id is required.');
+      throw new errors.BadRequestError('id is required.');
+    }
+    const user = _.pick(body, ['name', 'dob', 'address', 'description']);
```

Comment:
`name` is required on creation, but it seems update operation doesn't have such check.

Reply:
Thanks for your comments.

In my design, `id ` is the primary key for user model, user can update any other field by id, so `name ` is not required on update operation. Any concern, please let me know.

Reply:
If there is no `name` field in request body, name will not be updated in this operation.

Reply:
I mean the create api forbids `name: null` whilte update api allows `name: null`